fieldservice_agreement/models/agreement.py

```diff
@@ -58,3 +62,13 @@ def action_view_fsm_equipment(self):
             else:
                 action['domain'] = [('id', 'in', equipment_ids.ids)]
             return action
+
+    @api.multi
+    def _set_service_ids(self):
+        for agree in self:
+            servpros = self.env['agreement.serviceprofile'].\
+                search([('agreement_id', '=', self.name)])
+            ids = []
+            for ser in servpros:
+                ids.append(ser.id)
+            self.serviceprofile_ids = ids
```

Comment:
@osi-scampbell You are creating a new agreement. How can you set the service profiles by default based on the service profiles related to that agreement? It's a new one so there is no service profile that are linked to it yet.

Reply:
I think the confusion was I had serviceprofiles as Many2many when they should have been One2many. That method worked fine for Many2many but they should be One2many, and if they are then this function is not necessary.